nipype/interfaces/afni/utils.py

```diff
@@ -617,11 +617,15 @@ class CatMatvec(AFNICommand):
 
     def _format_arg(self, name, spec, value):
         if name == 'in_file':
-            return spec.argstr % (' '.join([i[0] + ' -' + i[1]
-                                            for i in value]))
+            xfm_args=''
+            for v in value:
+                if len(v[1])>0:
+                    xfm_args += ' ' +  v[0] + ' -' + v[1] + ' '
+                else:
+                    xfm_args += ' ' + v[0] + ' '
+            return spec.argstr % (xfm_args)
```

Comment:
I think it would be clearer to continue writing this as a comprehension, using the `x if y else z` idiom, but we can use more verbose variable names to make the logic clearer. I also think it would be useful to include a comment:

```Python
# Concatenate a series of filenames, with optional opkeys
return ' '.join('%s -%s' % (mfile, opkey) if opkey else mfile
                for mfile, opkey in value)
```

I dropped the initial `spec.argstr %`, just because `spec.argstr` is `"%s"`, so there's no real point, but you could keep it if you wanted.

Reply:
I agree this is more elegant, I changed it and confirmed it passed my test for formatting.